### PR TITLE
BAU: fix ci-pr-test failures

### PR DESCRIPTION
**Pull request by katstevens**
## What
The `pr-ci/ci-pr-test` job has been failing consistently for some weeks (see [an example failed build](https://cd.gds-reliability.engineering/teams/pay-dev/pipelines/pr-ci/jobs/ci-pr-test/builds/47.1)).

This job has been using [pipecleaner](https://github.com/alphagov/paas-cf/tree/main/tools/pipecleaner#features) to run several checks:
- Concourse's own validation
- Shellcheck on any shell or bash scripts
- Resource check, to flag any unused resources in a job description
- (Pipecleaner can also run Rubocop, but we've disabled that option)

I don't think the repo-renaming itself caused the failures, however we can't check further back as the old `omnibus-pr-test` pipeline no longer exists on Big Concourse.

## How
I've gone by the shellcheck suggestions, and validated those using the [online shellcheck tool](https://www.shellcheck.net/), as my local version of `pipecleaner` wasn't working for some reason. 

I've used `bash` instead of `sh` to resolve most of the errors - we're using for some tasks already, so may as well be consistent.

I added two 'ignores' for shellcheck, in `pr.yml` and `tasks/publish-pacts.yml` - not because they are unfixable, but because they are just time-consuming enough to get bogged down in, and I suspect they're actually are fine in context. I am willing to be persuaded!

## Not included in this PR
I haven't fixed the deprecation warnings - there are a lot of them and this PR is already full of small fiddly changes.

### CODE REVIEW

File: ci/pipelines/pr.yml [hunk 1]
```diff
@@ -1387,11 +1386,12 @@ jobs:
         outputs:
           - name: pacts
         run:
-          path: sh
+          path: bash
           dir: src
           args:
             - -ec
             - |
+              # shellcheck disable=SC1091
```
Comment (oswaldquek) [inline ci/pipelines/pr.yml:1394]: do we actually need this comment?

File: ci/tasks/publish-pacts.yml [hunk 2]
```diff
@@ -24,14 +24,16 @@ run:
       version="$(cat src/.git/resource/head_sha)"
       pr="$(cat src/.git/resource/pr)"
       cd pacts || exit 1
+
+      # shellcheck disable=SC2010
```
Comment (oswaldquek) [inline ci/tasks/publish-pacts.yml:28]: do we need this comment?

File: ci/pipelines/pr.yml (hunk 1 above)
Comment (katstevens) [reply ci/pipelines/pr.yml:1394]: Yep, this comment is what skips the shellcheck failure. Unless we can fix it instead?

Comment (oswaldquek) [reply ci/pipelines/pr.yml:1394]: Nah, it's ok. I'll approve the PR.

**Review by oswaldquek — APPROVED** [review]
LGTM

**Comment by katstevens** [comment]
This solved the shellcheck and resource failures, however the Concourse validation warnings are still causing the pipeline to fail (despite being just warnings!).

**Pull request merged**